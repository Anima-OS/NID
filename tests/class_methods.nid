
class Rect {
  int x(int a, int b);
  Rect* create(int x, int y, int w, int h);
};
