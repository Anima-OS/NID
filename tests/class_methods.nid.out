{ declarations: 
   [ { class: 
        { name: 'Rect',
          declarations: 
           [ { function: 
                { type: { name: 'int' },
                  name: 'x',
                  attributes: {},
                  parameters: 
                   [ { type: { name: 'int' }, name: 'a', attributes: {} },
                     { type: { name: 'int' }, name: 'b', attributes: {} } ] } },
             { function: 
                { type: { name: 'Rect', pointer: '*' },
                  attributes: {},
                  name: 'create',
                  parameters: 
                   [ { type: { name: 'int' }, name: 'x', attributes: {} },
                     { type: { name: 'int' }, name: 'y', attributes: {} },
                     { type: { name: 'int' }, name: 'w', attributes: {} },
                     { type: { name: 'int' }, name: 'h', attributes: {} } ] } } ] } } ] }
